### DISCUSSION TITLE
Differences in licensing Android and iOS SDKs

### DISCUSSION THREAD

**Discussion by wzieba** [Unleash/unleash#8557] [Q&A]
hi there 👋 I've noticed that Android SDK is released under Apache 2.0 ([link](https://github.com/Unleash/unleash-android/blob/a1093deb20dd32785f33d5b063c75b8051538f3a/unleashandroidsdk/build.gradle.kts#L104-L107)) but iOS SDK under MIT ([link](https://github.com/growthbook/growthbook-swift?tab=MIT-1-ov-file#readme)).

We're considering using Unleash in our projects, but Apache 2.0 license is a blocker for us, as it's not compatible with GPL-2.0.

I have 2 question:
- what's the reason to use Apache 2.0 for Android SDK?
- if there are no blockers, would you consider updating Android SDK's license to MIT, the same like iOS?

**Comment by chriswk** [comment]
Hi.

1. The main reason was that our main repo unleash/unleash is also Apache 2.0, so was assumed by our developers to be our default license.
2. We'll double check the licenses of our dependencies and if no blockers can be found, we're happy to relicense it as MIT.

**Reply by chriswk** [reply]
I made https://github.com/Unleash/unleash-android/issues/95 to make sure we follow up

**Comment by gastonfournier** [comment]
Hi @wzieba we don't have anything licensed under MIT. The swift repository you linked is from a different company.

Why is it a requirement for you to be GPL-2.0 compliant? It's the first time in 10 years we get a request like this, so we'd like to understand why this change is needed for you.

**Reply by wzieba** [reply]
> The swift repository you linked is from a different company.

Sorry! I was researching different solutions and confused links.

> Why is it a requirement for you to be GPL-2.0 compliant?

Several of our projects at Automattic, including the WordPress and WooCommerce mobile apps for both Android and iOS, are released under the GPL-2.0 license. Consequently, the licenses of our dependencies must also be compatible with GPL-2.0 requirements.

**Reply by gastonfournier** [reply]
Thanks @wzieba we're going to look into this and make sure we can make this change safely. Thanks for the feedback!

**Comment by gastonfournier** [comment]
Ok, both were on Apache 2.0:
- Swift: https://github.com/Unleash/unleash-proxy-client-swift/blob/5ed2bb057440d18dfe4b60ff44c38a57089e7f40/LICENSE
- Android: https://github.com/Unleash/unleash-android/blob/d1e0e3e158ed54f537991becdecce52625d5214f/unleashandroidsdk/build.gradle.kts#L112-L113

But we just released [Unleash Android SDK 2.0.0](https://github.com/Unleash/unleash-android/releases/tag/v2.0.0) with the MIT license, as having Apache 2.0 was not a conscious choice. Now we should consider if we're going to do the same with the Swift client.
